app/src/test/java/org/grammatek/simacorrect/spellcheckerservice/YfirlesturAnnotationTest.kt

```diff
@@ -0,0 +1,219 @@
+        val response = api.correctApiPost(text)
+        val dictionary = arrayListOf<String>()
+
+        val ylAnnotation = YfirlesturAnnotation(response, text)
+        var suggestionInfo = ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        for (suggestions in suggestionInfo) {
+            assertThat(suggestions.suggestionsCount).isAtMost(suggestionLimit)
+        }
+
+        suggestionLimit = 1
+        suggestionInfo = ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        for (suggestions in suggestionInfo) {
+            assertThat(suggestions.suggestionsCount).isAtMost(suggestionLimit)
+        }
+    }
+
+    @Test
+    fun `multiple whitespaces don't throw off indices for annotations`() {
+        val suggestionLimit = 5
+        val text = "   Afhverju  er   það  eki  hægt"
+        val startIndices = listOf(3, 23)
+        val endIndices = listOf(10, 25)
+        val dictionary = arrayListOf<String>()
+        val response = api.correctApiPost(text)
+        val ylAnnotation = YfirlesturAnnotation(response, text)
+        ylAnnotation.getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        // TODO: check for annotated words that are "suggestionless" as well.
+        for (i in ylAnnotation.suggestionsIndices.indices) {
+            assertThat(ylAnnotation.suggestionsIndices[i].startChar).isEqualTo(startIndices[i])
+            assertThat(ylAnnotation.suggestionsIndices[i].endChar).isEqualTo(endIndices[i])
+        }
+    }
+
+    @Test
+    fun `capitalization for the first word of a sentence to be ignored by Yfirlestur`() {
+        val suggestionLimit = 5
+        val capitalizedText = "Mig langar ekki út."
+        val nonCapitalizedText = "mig langar ekki út."
+        val dictionary = arrayListOf<String>()
+        val response = api.correctApiPost(capitalizedText)
+        // The spell checker service takes care of capitalizing the first letter
+        // so we have to test it differently e.g. see if the same sentence has
+        // different suggestions if the first word is capitalized or not.
+        val capitalizedSuggestionsInfo = YfirlesturAnnotation(response, capitalizedText).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+        val nonCapitalizedSuggestionsInfo = YfirlesturAnnotation(response, nonCapitalizedText).getSuggestionsForAnnotatedWords(suggestionLimit, dictionary)
+
+        assertThat(capitalizedSuggestionsInfo).isEqualTo(nonCapitalizedSuggestionsInfo)
+    }
+
+    @Test
+    fun `special characters don't throw off annotation indices`() {
+        // Yfirlestur's python binary is compiled with different support than of Android.
+        // As a consequence, Yfirlestur and Android count codepoints differently i.e. some
+        // special characters have different lengths in Yfirlestur, from Android.
+        val suggestionLimit = 0
+        val text = "\uD83E\uDD76\u200B er kallt og \uD83E\uDD75 er heittt"
```

Comment:
also here: this just tests a very special case, not the general functionality

Reply:
This is still a valuble test in my opinion, The name of the test might need to be renamed but this test helps determine if unicode characers contain "surrogate pairs" which throw of the character indices. This can't be changed to take data from an input file and adding multiple cases wouldn't be meaningful. 
Maybe renaming the test case would cause less confusion, renaming to something like "unicode surrogate pairs don't throw off character indices"